be/src/exec/hdfs_scanner_text.cpp

```diff
@@ -253,80 +249,70 @@ Status HdfsTextScanner::parse_csv(int chunk_size, ChunkPtr* chunk) {
             RETURN_IF_ERROR(_create_or_reinit_reader());
             continue;
         } else if (!status.ok()) {
-            LOG(WARNING) << "Status is not ok: " << status.to_string();
+            LOG(WARNING) << "Parse csv file failed: " << status;
             return status;
         }
 
-        fields.clear();
-        _reader->split_record(record, &fields);
-
-        if (!validate_utf8(record.data, record.size)) {
+        bool validate_res = validate_utf8(record.data, record.size);
+        if (!validate_res && options.invalid_field_as_null) {
+            LOG(WARNING) << "Face csv invalidate UTF-8 character line, append default value for this line";
+            chunk->get()->append_default();
             continue;
+        } else if (!validate_res) {
+            return Status::InternalError("Face csv invalidate UTF-8 character line.");
         }
 
-        bool has_error = false;
-        std::string error_msg;
-        int num_materialize_columns = _scanner_params.materialize_slots.size();
-        int field_size = fields.size();
-        if (_scanner_params.hive_column_names->size() != field_size) {
-            VLOG(7) << strings::Substitute("Size mismatch between hive column $0 names and fields $1!",
-                                           _scanner_params.hive_column_names->size(), fields.size());
-        }
+        CSVReader::Fields fields{};
+        _reader->split_record(record, &fields);
+
+        size_t num_materialize_columns = _scanner_params.materialize_slots.size();
+
+        // Fill materialize columns first, then fill partition column
         for (int j = 0; j < num_materialize_columns; j++) {
-            // TODO slot maybe null?
-            const auto slot = _scanner_params.materialize_slots[j];
+            const auto& slot = _scanner_params.materialize_slots[j];
             DCHECK(slot != nullptr);
 
-            std::string col_name = _scanner_params.materialize_slots[j]->col_name();
-            if (!_scanner_params.case_sensitive) {
-                std::transform(col_name.begin(), col_name.end(), col_name.begin(), ::tolower);
-            }
-            int index = _scanner_params.materialize_index_in_chunk[j];
-            int column_field_index = _columns_index[_scanner_params.materialize_slots[j]->col_name()];
-            Column* column = _column_raw_ptrs[index];
-            if (column_field_index < field_size) {
-                const Slice& field = fields[column_field_index];
+            const std::string formatted_slot_name =
```

Comment:
extract as a function, I think both parquet and orc should benefit from it.

Reply:
```
class SlotVisitor {
    SlotVisitor(materialize_slots)

    const std::string column_name();
}
```

Reply:
parquet and orc's logic is slightly different. And I don't want this pr become too large.